testing/integration/otel_test.go

```diff
@@ -1595,3 +1597,323 @@ service:
+                fields:
+                  id: 0879f47d-df41-464d-8462-bc2b8fef45bf
+                target: agent
+            - drop_event:
+                when:
+                  regexp:
+                    component:
+                      id: .*-monitoring$
+            - drop_event:
+                when:
+                  regexp:
+                    message: ^Non-zero metrics in the last
+            - copy_fields:
+                fields:
+                  - from: data_stream.dataset
+                    to: data_stream.dataset_original
+            - drop_fields:
+                fields:
+                  - data_stream.dataset
+            - copy_fields:
+                fail_on_error: false
+                fields:
+                  - from: component.dataset
+                    to: data_stream.dataset
+                ignore_missing: true
+            - copy_fields:
+                fail_on_error: false
+                fields:
+                  - from: data_stream.dataset_original
+                    to: data_stream.dataset
+            - drop_fields:
+                fields:
+                  - data_stream.dataset_original
+                  - event.dataset
+            - copy_fields:
+                fields:
+                  - from: data_stream.dataset
+                    to: event.dataset
+            - drop_fields:
+                fields:
+                  - ecs.version
+                ignore_missing: true
+            - add_formatted_index:
+                index: '%{[data_stream.type]}-%{[data_stream.dataset]}-%{[data_stream.namespace]}'
+    output:
+      otelconsumer:
+    path:
+      data: {{.HomeDir}}/filestream-monitoring
+    queue:
+      mem:
+        flush:
+          timeout: 0s
+    logging:
+      level: info
+      selectors:
+        - '*'
+    processors:
+      - add_host_metadata:
+          when:
+            not:
+              contains:
+                tags: forwarded
+      - add_cloud_metadata: ~
+      - add_docker_metadata: ~
+      - add_kubernetes_metadata: ~
+    setup.ilm.enabled: false 
+    setup.template.enabled: false
+    filebeat.config.modules.enabled: false
+    http.enabled: true
+    http.host: {{ .SocketEndpoint }}
+exporters:
+  debug:
+    use_internal_logger: false
+    verbosity: detailed
+  elasticsearch/log:
+    endpoints:
+      - {{.ESEndpoint}}
+    compression: none
+    api_key: {{.ESApiKey}}
+    logs_dynamic_index:
+      enabled: true
+    batcher:
+      enabled: true
+      flush_timeout: 1s
+    mapping:
+      mode: bodymap	  
+service:
+  pipelines:
+    logs:
+      receivers:
+        - filebeatreceiver/filestream-monitoring-agent
+      exporters:
+        - elasticsearch/log
+`
+
+	beatsApiKey, err := base64.StdEncoding.DecodeString(esApiKey.Encoded)
+	require.NoError(t, err, "error decoding api key")
+
+	var configBuffer bytes.Buffer
+	require.NoError(t,
+		template.Must(template.New("config").Parse(configTemplate)).Execute(&configBuffer,
```

Comment:
`template.Must` will panic on an error, you don't need it as a part of this call